src/Features/Core/Portable/LanguageServices/SymbolDisplayService/SymbolDisplayClassificationHelper.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DocumentationComments;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Classification;
+
+namespace Microsoft.CodeAnalysis.LanguageServices
+{
+    internal static class SymbolDisplayClassificationHelper
+    {
+        public static ImmutableArray<SymbolDisplayPart> ConvertClassifications(
+            SourceText sourceText, int startPosition, IEnumerable<ClassifiedSpan> classifiedSpans, bool insertSourceTextInGaps = false)
+        {
+            var parts = new List<SymbolDisplayPart>();
+
+            foreach (var span in classifiedSpans)
+            {
+                // If there is space between this span and the last one, then add a space.
+                if (startPosition != span.TextSpan.Start)
+                {
+                    if (insertSourceTextInGaps)
+                    {
+                        parts.Add(new SymbolDisplayPart(SymbolDisplayPartKind.Text, null,
+                            sourceText.ToString(TextSpan.FromBounds(
+                                startPosition, span.TextSpan.Start))));
+                    }
+                    else
+                    {
+                        parts.AddRange(Space());
+                    }
+                }
+
+                var kind = GetDisplayPartKind(span.ClassificationType);
+                parts.Add(new SymbolDisplayPart(kind, null, sourceText.ToString(span.TextSpan)));
+                startPosition = span.TextSpan.End;
+            }
+
+            return parts.ToImmutableArray();
```

Comment:
Since we allocate a `List<SymbolDisplayPart>` just to allocate an `ImmutableArray<SymbolDisplayPart>` (which internally allocates a new `SymbolDisplayPart[]` at the end, would it be better just to use an `ImmutableArray<SymbolDisplayPart>.Builder`?

```C#
var parts = ImmutableArray.CreateBuilder<SymbolDisplayPart>();

...

return parts.ToImmutable();
```